packages/flutter/lib/src/widgets/fade_in_image.dart

```diff
@@ -534,23 +537,14 @@ class _AnimatedFadeOutFadeInState extends ImplicitlyAnimatedWidgetState<_Animate
         weight: widget.fadeInDuration.inMilliseconds.toDouble(),
       ),
     ]));
-    if (!widget.isTargetLoaded && _isValid(_placeholderOpacity!) && _isValid(_targetOpacity!)) {
```

Comment:
why remove this logic?

Reply:
as the comment suggests, this code jumps back to the placeholder image for the "new" target image.

i.e., `old image --[user changes the image]--> jump to placeholder --[wait for new image]--> fade in new image.`

but according to image gapless playback, the placeholder will never be shown again, the old image will act as a placeholder and the new image replace the old one w/o "fade in fade out".